### PR TITLE
Fix JFR recording for Hyperfoil

### PR DESCRIPTION
**Pull request by jabolina**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,6 +101,19 @@ All of the roles internally have the notion of the following "steps" that are co
 
 Running a playbook will run the respective roles for each of these. If instead you want to run a subset you can use the operation argument as such `ansible-playbook -e operation=run`.
 
+#### JFR Hyperfoil
+
+To record Hyperfoil into JFR files we need to enable the options. It is possible to record on the controller and the agents separately, which with a different approach.
+To collect on the controller, when running the controller benchmark to start, pass the `-e hyperfoil_jfr=true` argument.
```
Comment (jabolina) [inline README.md:107]: I could only get this working with the change on https://github.com/Hyperfoil/hyperfoil_setup/pull/6.

File: README.md [hunk 2]
```diff
@@ -101,6 +101,19 @@ All of the roles internally have the notion of the following "steps" that are co
...
+
+The agents are started by the controller. Therefore, we need to provide the custom JVM arguments, we can do it when running the agent playbook. This is done with `-e agent_java_args="'-Xmx800m -XX:+FlightRecorder -XX:StartFlightRecording=compress=false,delay=10s,duration=24h,settings=profile,filename=/tmp/hyperfoil/workspace/hfa-record.jfr'"` argument.
```
Comment (wburns) [inline README.md:109]: The java_args here are all optional and the filename is only required if `-e download_jfr=true` is also specified. My point being is that we may want to explain they can use any flight recorder settings but just the filename has to be very specific if downloaded.

File: README.md (hunk 1 above)
Comment (wburns) [inline README.md:107]: Are you sure this didn't work for the agent as well? I seem to remember that it used to generate a jfr on the agent, correct me if I am wrong though.

File: roles/hyperfoil_controller/tasks/retrieve_stats.yml [hunk 3]
```diff
@@ -15,7 +15,7 @@
   when: download_results
 - name: Download JFR
   ansible.builtin.fetch:
-    src: "{{ hyperfoil_dir }}/hfc-{{ ansible_hostname }}.jfr"
+    src: "/tmp/hyperfoil/hfc-ip-{{ ansible_default_ipv4.address | replace('.', '-') }}.jfr"
```
Comment (wburns) [inline roles/hyperfoil_controller/tasks/retrieve_stats.yml:18]: The more I look at this, we probably should reuse the hyperfoil_dir like we the hyperfoil ansible scripts do and just default it to /tmp/hyperfoil

File: README.md (hunk 2 above)
Comment (jabolina) [reply README.md:109]: Sure, let me update it.

File: README.md (hunk 1 above)
Comment (jabolina) [reply README.md:107]: Yeah, I couldn't get it to collect on the agents other than the java_args. The https://github.com/Hyperfoil/hyperfoil_test doesn't have an option for JFR and the deployment (https://github.com/Hyperfoil/Hyperfoil/blob/master/clustering/src/main/java/io/hyperfoil/deploy/ssh/SshDeployedAgent.java#L96) also doesn't pass the options from the controller to the agents.

File: roles/hyperfoil_controller/tasks/retrieve_stats.yml [hunk 4]
```diff
@@ -15,7 +15,7 @@
   when: download_results
 - name: Download JFR
   ansible.builtin.fetch:
-    src: "{{ hyperfoil_dir }}/hfc-{{ ansible_hostname }}.jfr"
+    src: "{{ hyperfoil_dir | default('/tmp/hyperfoil') }}/hfc-ip-{{ ansible_default_ipv4.address | replace('.', '-') }}.jfr"
```
Comment (wburns) [inline roles/hyperfoil_controller/tasks/retrieve_stats.yml:18]: Think there might have been a disconnect with what I meant about having the default be /tmp/hyperfoil. I meant to set the `hyperfoil_dir` variable in https://github.com/infinispan/infinispan-ansible-benchmark/blob/main/group_vars/all.yaml to be /tmp/hyperfoil this way it is picked up by all of the roles. Then we can remove all usages of /tmp/hyperfoil and replace it with {{ hyperfoil_dir }}

Comment (jabolina) [reply roles/hyperfoil_controller/tasks/retrieve_stats.yml]: Updated.

File: roles/hyperfoil_controller/tasks/retrieve_stats.yml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 - name: Find number of requests
   community.general.read_csv:
-    path: "{{ '/tmp/hyperfoil/workspace/run/' + test_runid + '/stats/total.csv' }}"
+    path: "{{ hyperfoil_dir }} + {{ '/workspace/run/' + test_runid + '/stats/total.csv' }}"
```
Comment (wburns) [inline roles/hyperfoil_controller/tasks/retrieve_stats.yml:3]: This is actually broken, shouldn't be the spaces or + at all.

File: roles/hyperfoil_controller/tasks/retrieve_stats.yml [hunk 6]
```diff
@@ -9,13 +9,13 @@
   loop: "{{ test_requests.dict.keys()|list }}"
 - name: Download Results
   ansible.builtin.fetch:
-    src: "{{ '/tmp/hyperfoil/workspace/run/' + test_runid + '/stats/total.csv' }}"
+    src: "{{ hyperfoil_dir }} + {{ '/workspace/run/' + test_runid + '/stats/total.csv' }}"
```
Comment (wburns) [inline roles/hyperfoil_controller/tasks/retrieve_stats.yml:12]: Same here no space or +.

File: roles/hyperfoil_controller/tasks/retrieve_stats.yml (hunk 5 above)
Comment (wburns) [reply roles/hyperfoil_controller/tasks/retrieve_stats.yml:3]:
```
TASK [hyperfoil_controller : Find number of requests] **************************

fatal: [ec2-3-16-137-142.us-east-2.compute.amazonaws.com -> ec2-3-138-61-105.us-east-2.compute.amazonaws.com]: FAILED! => {"changed": false, "msg": "Unable to open file: [Errno 2] No such file or directory: '/tmp/hyperfoil + /workspace/run/0001/stats/total.csv'"}
```

**Pull request merged**